crates/sui-graphql-rpc/src/server/data_provider.rs

```diff
@@ -88,8 +93,11 @@ pub(crate) async fn fetch_owned_objs(
         .await?;
 
     pg.data.iter().try_for_each(|n| {
-        if n.error.is_some() || n.data.is_none() {
-            return Err(Error::new("error"));
+        // TODO (wlmyng): support partial data with union error type
```

Comment:
@oxade and I talked briefly about this, and the conclusion of that conversation was that we probably don't want partial failures in the middle of a cursor connection, but on reflection, maybe we do, because this connection is also the only way we have to do an object multi-get.

Incidentally, now that we don't have to fancy binary op queries, we'll need to expose multi-get some other way (not related to this PR, just recording that so we don't forget -- I'll update the schema for this shortly).

Reply:
Multi-get in the interface: https://github.com/MystenLabs/sui/pull/13404

Reply:
Perhaps another point in favor of partial failures, if I try to multi-get objects and some of them don't exist anymore/ other client-fault errors, I would need to query n times until I can prune out the problematic ones or perhaps give up and just make individual calls to gather the data. Returning partial success and error responses would reduce this painpoint

Reply:
Okay I understand now. Yeah it does make sense to not allow partial failures in the middle of a cursor connection. But separately I think it'd be nice to return partial results for a multi-get

Reply:
Lol sorry for the monologuing, but since each object in the connection has a cursor, at least to me it doesn't actually seem to be an issue to try to return as much data as we can https://github.com/MystenLabs/sui/pull/13394
We can do that + let them know, or the caller might implicitly pick up that the cursor they're using should be invalid

Reply:
Yes, I agree :) My comment was meant to indicate that before we thought we shouldn't do this, but at least my mind is changing/changed on it.